File: python/queues.py
if __name__ == "__main__":
    queue = []  # Queues use FIFO (First In, First Out)

    # Enqueue
    queue.append(10)
    queue.append(12)
    queue.append(9)
    queue.append(8)
    queue.append(3)

    print(f"Queue: {queue}")

    print(f"The queue has {len(queue)} elements.")
    print("\n")

    # Front and Back Elements
    print(f"The front element is: {queue[0]}")
    print(f"The back element is: {queue[-1]}")
    print("\n")

    # Dequeue
    queue.pop(0)
    queue.pop(0)

    print(f"(After dequeue) Queue: {queue}")
    print(f"The front element is: {queue[0]}")
    print(f"The back element is: {queue[-1]}")

File: c++/queues.cpp
#include <iostream>
#include <queue>

int main() {
    std::queue<int> queue; // Queues use FIFO (First In, First Out)

    // Enqueue
    queue.push(10);
    queue.push(12);
    queue.push(9);
    queue.push(8);
    queue.push(3);

    std::cout << "The queue has " << queue.size() << " elements." << "\n";
    std::cout << "\n";

    // Front and Back Elements
    std::cout << "The front element is: " << queue.front() << "\n";
    std::cout << "The back element is: " << queue.back() << "\n";
    std::cout << "\n";

    // Dequeue
    queue.pop();
    queue.pop();

    std::cout << "(After dequeue) The front element is: " << queue.front() << "\n";
    std::cout << "(After dequeue) The back element is: " << queue.back() << "\n";

    return 0;
}